lib/tapioca/dsl/compilers/active_record_relations.rb

```diff
@@ -376,7 +376,7 @@ def create_group_chain_methods(klass)
             return_type: "T.self_type",
           )
 
-          CALCULATION_METHODS.each do |method_name|
+          (CALCULATION_METHODS + [:size]).each do |method_name|
```

Comment:
Is there any reason to not add `:size` into the `CALCULATION_METHODS` array itself?

Reply:
Hey Mark!

> where the method name doesn't fit nicely into into an array and the method should be defined outside of the scope of the case statement looped over an array.

That seems reasonable to me.

> If you give an argument like User.where(foo: 1) you do not in fact get to use the methods on the PrivateAssociationRelationWhereChain returned.

I can't test this right now, could you explain the issue?

> And now I can fully appreciate how difficult the job is of the Tapioca team!

Haha thanks! Trying to tame Ruby meta-programming with types is a formidable challenge :)

Reply:
Regarding this:

> And when looking into how the .to_a worked, I'm now finding that the GeneratedRelationMethods#where chain typing is actually incorrect. If you give an argument like User.where(foo: 1) you do not in fact get to use the methods on the PrivateAssociationRelationWhereChain returned. Additionally, you can't call :to_a on an object that is a where chain (User.where.not.to_a), unlike what the generated rbi suggests.

Ufuk has opened a PR to make "where chains" no longer be relations: #553048